### PR TITLE
fix(config): split lzw31-sn param 16 normalize param names

### PR DESCRIPTION
**Pull request by mattsch**
This makes all the following changes:
* Param 16 split out into separate parameters
* Update all the param names for the LED effects across the three device types for consistency
* Set defaults for duration and brightness
* Update lzw36 to include a selector for the brightness to match the others

### CODE REVIEW

**Comment by mattsch** [comment]
I was also considering adding the duration options like Inovelli has for their [Hubitat](https://github.com/InovelliUSA/Hubitat/blob/master/Drivers/inovelli-switch-red-series-lzw30-sn.src/inovelli-switch-red-series-lzw30-sn.groovy#207) and [SmartThings](https://github.com/InovelliUSA/SmartThingsInovelli/blob/935149ee19ab807378a9906cf70a778e92e9dbe7/devicetypes/inovelliusa/inovelli-dimmer-red-series-lzw31-sn.src/inovelli-dimmer-red-series-lzw31-sn.groovy#L299). Any objections @AlCalzone?

**Comment by AlCalzone** [comment]
> I was also considering adding the duration options like Inovelli has for their [Hubitat](https://github.com/InovelliUSA/Hubitat/blob/master/Drivers/inovelli-switch-red-series-lzw30-sn.src/inovelli-switch-red-series-lzw30-sn.groovy?rgh-link-date=2021-03-14T14%3A04%3A16Z#207) and [SmartThings](https://github.com/InovelliUSA/SmartThingsInovelli/blob/935149ee19ab807378a9906cf70a778e92e9dbe7/devicetypes/inovelliusa/inovelli-dimmer-red-series-lzw31-sn.src/inovelli-dimmer-red-series-lzw31-sn.groovy#L299). Any objections @AlCalzone?

Not as single options please. These can stay as manual entry with a corresponding description.

File: packages/config/config/devices/0x031e/lzw36.json [hunk 1]
```diff
@@ -456,8 +502,8 @@
 			]
 		},
 		"25[0x7f000000]": {
-			"label": "Fan LED Strip Effect",
-			"description": "Fan LED Strip Effect. 0 = Off, 1 = Solid, 2 = Slow Blink, 3 = Fast Blink, 4 = Chase, 5 = Pulse Default: 0",
+			"label": "Fan LED Effect Type",
+			"description": "Fan LED Strip Effect Type. 0 = Off, 1 = Solid, 2 = Slow Blink, 3 = Fast Blink, 4 = Chase, 5 = Pulse Default: 0",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]: That description does not add value over what is defined below. Please remove.

File: packages/config/config/devices/0x031e/lzw36.json [hunk 2]
```diff
@@ -494,31 +540,77 @@
 			]
 		},
 		"25[0xff0000]": {
-			"label": "Fan LED Strip Effect Duration",
+			"label": "Fan LED Effect Duration",
 			"description": "Fan LED Strip Duration. 1 to 60 = seconds, 61 to 120 minutes, 121 - 254 = hours, 255 = Indefinitely Default: 0",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]:
```suggestion
			"description": "Fan LED Strip Duration. 1 to 60 = seconds, 61 to 120 minutes, 121 - 254 = hours, 255 = Indefinitely",
```
There is some discrepancy here. The old description said default is 0, you define 255 below. And if 255 is correct, 0 has no meaning.

File: packages/config/config/devices/0x031e/lzw36.json [hunk 3]
```diff
@@ -494,31 +540,77 @@
...
 			"unsigned": true,
 			"readOnly": false,
 			"writeOnly": false,
 			"allowManualEntry": true
 		},
 		"25[0xff00]": {
-			"label": "Fan LED Strip Effect Intensity",
-			"description": "Fan LED Strip Intensity 0 to 9. 0 = dim, 9 = bright Default: 3",
+			"label": "Fan LED Effect Brightness",
+			"description": "Fan LED Strip Brightness 0 to 10. 0 = dim, 10 = bright Default: 3",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]: Description seems unnecessary. Remove please.

Comment (AlCalzone) [reply packages/config/config/devices/0x031e/lzw36.json]: Please also check the other parameters that are similar.

File: packages/config/config/devices/0x031e/lzw36.json (hunk 2 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x031e/lzw36.json]: Please also check the other parameters that are similar.

File: packages/config/config/devices/0x031e/lzw36.json (hunk 1 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x031e/lzw36.json]: Please also check the other parameters that are similar.

File: packages/config/config/devices/0x031e/lzw36.json [hunk 4]
```diff
@@ -386,31 +386,77 @@
...
+				{
+					"label": "100%",
+					"value": 10
+				}
+			]
 		},
 		"24[0xff]": {
-			"label": "Light LED Strip Effect Color",
+			"label": "Light LED Effect Color",
 			"description": "Light LED Strip Effect Color. Color - 0 - 255. Hue color wheel. value/255 * 360 = Hue color wheel value Range: 0-255 Default: 0",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]: While we're cleaning up, this description should be simplified - it is a bit redundant in itself. Please no range/default in descriptions.

Please also check the other parameters that are similar.

**Review by AlCalzone — CHANGES_REQUESTED** [review]
A few things... and since you touched these files (sorry 😉), I think we should consolidate the different LZW31 files using conditional parameters now.

See https://zwave-js.github.io/node-zwave-js/#/config-files/conditional-settings for details. Might also be worth to check out our brand-new [style guide](https://zwave-js.github.io/node-zwave-js/#/config-files/style-guide)

**Comment by mattsch** [comment]
@zwave-js-bot fix lint

**Comment by mattsch** [comment]
@AlCalzone So it should all be consistent. May be wrong, but at least it's all consistently wrong. 😄

File: packages/config/config/devices/0x031e/lzw30-sn.json [hunk 5]
```diff
@@ -352,6 +353,7 @@
 		"51": {
 			"$if": "firmwareVersion >= 1.19",
 			"label": "Instant On",
+			"description": "Controls the 700ms button delay. Note, if you disable the delay, it will also disable multi-tap scenes.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw30-sn.json]:
```suggestion
			"description": "Enabling this disables the 700ms button delay and multi-tap scenes.",
```
Seems a bit less confusing to me.

File: packages/config/config/devices/0x031e/lzw30-sn.json [hunk 6]
```diff
@@ -361,17 +363,18 @@
...
-					"label": "700ms Delay",
+					"label": "Disabled",
 					"value": 1
 				}
 			]
 		},
 		"8[0xff]": {
-			"label": "LED Strip Effect (Color)",
+			"label": "LED Effect Color",
+			"description": "Uses a scaled hue value (realHue/360 * 255).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw30-sn.json]:
```suggestion
			"description": "Uses a scaled hue value (realHue / 360 * 255).",
```

File: packages/config/config/devices/0x031e/lzw31-sn.json [hunk 7]
```diff
@@ -228,45 +286,319 @@
 		},
 		"13": {
 			"label": "LED Indicator Color",
+			"description": "Uses a scaled hue value (realHue/360 * 255).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-sn.json]:
```suggestion
			"description": "Uses a scaled hue value (realHue / 360 * 255).",
```

File: packages/config/config/devices/0x031e/lzw31-sn.json [hunk 8]
```diff
@@ -228,45 +286,319 @@
...
+					"label": "100%",
+					"value": 10
+				}
+			]
 		},
-		"16": {
-			"label": "LED Strip Effect",
+		"16[0xff]": {
+			"label": "LED Effect Color",
+			"description": "Uses a scaled hue value (realHue/360 * 255).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-sn.json]:
```suggestion
			"description": "Uses a scaled hue value (realHue / 360 * 255).",
```

File: packages/config/config/devices/0x031e/lzw31-sn.json [hunk 9]
```diff
@@ -367,8 +700,9 @@
 			"allowManualEntry": true
 		},
 		"51": {
-			"label": "Physical On/Off Delay",
-			"description": "The delay that occurs after pressing the physical button to turn the switch on/off is removed. Consequently this also removes the following scenes: 2x, 3x, 4x, 5x tap. 1x tap and config button scenes still work.",
+			"$if": "firmwareVersion >= 1.47",
+			"label": "Instant On",
+			"description": "Controls the 700ms button delay. Note, if you disable the delay, it will also disable multi-tap scenes.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw31-sn.json]:
```suggestion
			"description": "Enabling this disables the 700ms button delay and multi-tap scenes.",
```
Seems a bit less confusing to me.

File: packages/config/config/devices/0x031e/lzw36.json [hunk 10]
```diff
@@ -47,55 +49,90 @@
...
+					"value": 0
+				},
+				{
+					"label": "Synced with Parameter 1",
+					"value": 99
+				}
+			]
 		},
 		"3": {
 			"label": "Light Ramp Rate",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]: Please make this consistent with param 12/13

File: packages/config/config/devices/0x031e/lzw36.json [hunk 11]
```diff
@@ -47,55 +49,90 @@
...
+				{
+					"label": "Synced with Parameter 1",
+					"value": 99
+				}
+			]
 		},
 		"4": {
-			"label": "Light Ramp Rate (From Switch)",
-			"description": "This changes the speed in which the attached light turns on or off from the physical switch. For example, when a user presses the up or down button, this is the speed in which those actions take place. A setting of 0 should turn the light immediately on or off (almost like an on/off switch). Increasing the value should slow down the transition speed. A setting of 99 should keep this in sync with parameter 1. Range:0-99 Default: 99",
+			"label": "Light Ramp Rate From Switch",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]: Please make this consistent with param 12/13

File: packages/config/config/devices/0x031e/lzw36.json [hunk 12]
```diff
@@ -252,91 +296,324 @@
...
+				{
+					"label": "Last State",
+					"value": 100
+				}
+			]
 		},
 		"18": {
 			"label": "Light LED Indicator Color",
-			"description": "This is the color of the Light LED strip represented as part of the HUE color wheel. Since the wheel has 360 values and this parameter only has 255, the following equation can be used to determine the color: value/255 * 360 = Hue color wheel value Range: 0 to 255 Default: 170",
+			"description": "Uses a scaled hue value (realHue/360 * 255).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]:
```suggestion
			"description": "Uses a scaled hue value (realHue / 360 * 255).",
```

File: packages/config/config/devices/0x031e/lzw36.json [hunk 13]
```diff
@@ -252,91 +296,324 @@
...
+				{
+					"label": "100%",
+					"value": 10
+				}
+			]
 		},
 		"20": {
 			"label": "Fan LED Indicator Color",
-			"description": "This is the color of the Fan LED strip represented as part of the HUE color wheel. Since the wheel has 360 values and this parameter only has 255, the following equation can be used to determine the color: value/255 * 360 = Hue color wheel value Range: 0 to 255 Default: 170",
+			"description": "Uses a scaled hue value (realHue/360 * 255).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]:
```suggestion
			"description": "Uses a scaled hue value (realHue / 360 * 255).",
```

File: packages/config/config/devices/0x031e/lzw36.json [hunk 14]
```diff
@@ -386,32 +708,77 @@
 			]
 		},
 		"24[0xff0000]": {
-			"label": "Light LED Strip Effect Duration",
-			"description": "Light LED Strip Effect Duration. 1 to 60 = seconds, 61 to 120 minutes, 121 - 254 = hours, 255 = Indefinitely Default: 0",
+			"label": "Light LED Effect Duration",
+			"description": "1 to 60 = seconds, 61 to 120 = minutes, 121 - 254 = hours, 255 = indefinitely",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]:
```suggestion
			"description": "1 to 60 = seconds, 61 to 120 = minutes (minus 60), 121 - 254 = hours (minus 120), 255 = indefinitely",
```

File: packages/config/config/devices/0x031e/lzw36.json [hunk 15]
```diff
@@ -494,32 +860,77 @@
 			]
 		},
 		"25[0xff0000]": {
-			"label": "Fan LED Strip Effect Duration",
-			"description": "Fan LED Strip Duration. 1 to 60 = seconds, 61 to 120 minutes, 121 - 254 = hours, 255 = Indefinitely Default: 0",
+			"label": "Fan LED Effect Duration",
+			"description": "1 to 60 = seconds, 61 to 120 = minutes, 121 - 254 = hours, 255 = indefinitely",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]: See 24[0xff0000]

File: packages/config/config/devices/0x031e/lzw36.json [hunk 16]
```diff
@@ -386,32 +708,77 @@
...
+					"label": "100%",
+					"value": 10
+				}
+			]
 		},
 		"24[0xff]": {
-			"label": "Light LED Strip Effect Color",
-			"description": "Light LED Strip Effect Color. Color - 0 - 255. Hue color wheel. value/255 * 360 = Hue color wheel value Range: 0-255 Default: 0",
+			"label": "Light LED Effect Color",
+			"description": "Uses a scaled hue value (realHue/360 * 255).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]:
```suggestion
			"description": "Uses a scaled hue value (realHue / 360 * 255).",
```

File: packages/config/config/devices/0x031e/lzw36.json [hunk 17]
```diff
@@ -494,32 +860,77 @@
...
+					"label": "100%",
+					"value": 10
+				}
+			]
 		},
 		"25[0xff]": {
-			"label": "Fan LED Strip Effect Color",
-			"description": "Fan LED Color 0 - 255. Hue color wheel. value/255 * 360 = Hue color wheel value Range: 0-255 Default: 0",
+			"label": "Fan LED Effect Color",
+			"description": "Uses a scaled hue value (realHue/360 * 255).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]:
```suggestion
			"description": "Uses a scaled hue value (realHue / 360 * 255).",
```

File: packages/config/config/devices/0x031e/lzw36.json [hunk 18]
```diff
@@ -628,7 +1059,7 @@
 		},
 		"31": {
 			"label": "Local Protection",
-			"description": "Enable local protection on these buttons. 0 = none, 1 = light, 2 = fan, 3 = both.",
+			"description": "Enables for these buttons.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]:
```suggestion
```

File: packages/config/config/devices/0x031e/lzw36.json [hunk 19]
```diff
@@ -658,7 +1089,8 @@
 		},
 		"51": {
 			"label": "Instant On",
-			"description": "Enables instant on (ie: disables the 700ms button delay). Note, if you disable the delay, it will also disable scene control except for the following: Light on/off pressed, held, released, Fan on/off pressed, held, released & light up/down fan up/down pressed (firmware 1.36+).",
+			"$if": "firmwareVersion >= 1.36",
+			"description": "Controls the 700ms button delay. Note, if you disable the delay, it will also disable multi-tap scenes.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw36.json]: See other variants of this parameter where I proposed a diffent description.

File: packages/config/config/devices/0x031e/lzw30-sn.json [hunk 20]
```diff
@@ -84,6 +84,7 @@
 		},
 		"5": {
 			"label": "LED Indicator Color",
+			"description": "Uses a scaled hue value (realHue/360 * 255).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x031e/lzw30-sn.json]:
```suggestion
			"description": "Uses a scaled hue value (realHue / 360 * 255).",
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]
A few more things

**Comment by mattsch** [comment]
@AlCalzone I believe this should be ready to go.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**